client.go

```diff
@@ -0,0 +1,295 @@
+package socketio
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/googollee/go-socket.io/engineio"
+	"github.com/googollee/go-socket.io/engineio/transport"
+	"github.com/googollee/go-socket.io/engineio/transport/polling"
+	"github.com/googollee/go-socket.io/logger"
+	"github.com/googollee/go-socket.io/parser"
+)
+
+// Server is a go-socket.io server.
+type Client struct {
+	conn      *conn
+	namespace string
+	handlers  *namespaceHandlers
+	url       string
+	opts      *engineio.Options
+}
+
+// NewServer returns a server.
+func NewClient(uri string, opts *engineio.Options) (*Client, error) {
```

Comment:
it is so awful public API for client :( please looks [this](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) article

Reply:
Please fix it, PRs welcome :)